internal/snmp/translate.go

```diff
@@ -18,14 +19,21 @@ var m sync.Mutex
 var once sync.Once
 var cache = make(map[string]bool)
 
-func appendPath(path string) {
+type MibLoader interface {
+	loadModule(path string) error
+	appendPath(path string)
+}
+
+type GosmiMibLoader struct{}
```

Comment:
I don't see the added value for this? Could you explain?

Reply:
Sure! This is called dependency injection. It lets us have mock functions in tests so the tests won't call gosmi since we just want to check the logic of traversing the folders.

Reply:
Thanks. Maybe the struct could also be used for the lock,once and folder cache?

Reply:
They are used in the method `appendPath` and `loadModule`, but also in the function `walkPaths`. So if that `walkPaths` could be transformed to be also a method of `GosmiMibLoader`, then these snmp package vars could be moved into that struct..

Reply:
since `walkPaths` is what we want to test it cannot be added to the interface and mocked the same way as `appendPath` and `loadModule`. It doesn't seem important to but the mutex and cache in `gosmiMibLoader` when gosmi is a global library.